5.0 release

minuscorp opened the pull request:
Following the directions of the architectonical decisions made for the 5.1 release, we want this library for be usable for as many users as possible. That's why, we want to provide also a full Swift 5.0 support relying on RxSwift as a substitute for Combine.

sebastianvarela commented:
<!--
  0 Errors
  0 Warnings
  0 Messages
  1 Markdown
-->

## Danger has errored
[!] Invalid `Dangerfile` file: No such file or directory - ./output. Updating the Danger gem might fix the issue. Your Danger version: 6.0.6, latest Danger version: 6.0.9

```
 #  from Dangerfile:6
 #  -------------------------------------------
 #  
 >  bq_helpers.scan_files
 #  
 #  -------------------------------------------
```

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

r-pedraza commented on Sources/MiniSwift/Dispatcher.swift:
```diff
@@ -0,0 +1,234 @@
+
+import Foundation
+import RxSwift
+import NIOConcurrencyHelpers
```
oh! Cool! NIO framework 🤤

minuscorp commented:
In response of every point:

> The support of Carthage don't work, NIO don't support Carthage because don't have shared schemes.

Maybe the best option here is to drop Carthage support, discussable though.
 
> The name of the Product has changed from Mini to MiniSwift, I think that is more easy to use import Mini that import MiniSwift, Swift is redundant in this context.

The Cocoapods bundle still is being named and imported by `import Mini`, I'm changing the SPM to have the same name, my bad 🤦‍♂ .

> Similar to 2, the name of the xcodeproj has changed and the Sources folder have a unnecessary level called MiniSwift

Fixed!

> The protocol State, I think that should be renamed to StoreState because State is too generic.

Ain't protocol named `State`. There's a file named `State.swift`, renamed to `StateType.swift` to match its contents.

> We don't have Changeable behavior for store state anymore?

Not needed anymore 🎉 .

> This PR drop compatibility with fastlane utils (https://github.com/bq/fastlane-plugin-test_scheme) and danger plugin (https://github.com/bq/danger-bq_helpers), the result of the build is changed, I think that we need to see how to continue with this path or see how to support SPM without affecting most of the dependencies used. I think that we can update our plugins to use SPM.

I don't think that dropping dependencies is bad in this manner. As for now, I have updated the `Fastfile` to pass the tests in every platform and report the coverage to codecov.io, how the packages are created with SPM and how they are tested and compiled have changed and will change in further versions of SPM, so we'll need to keep this up-to-date.

> The LICENCE file is present, I propose to remove the license in every header because is redundant and lost focus in the important thing: the code

Is a common thing to copy one file or another, avoiding the license. It's a common OSS practice to add the license resume in the header, it is a small paragraph and it establishes that every source file is under the same license as the repo itself.

 > Rake is intended to resolve and prepare the project to build.. I think the swift build should be inside the rake script

I have updated the `Rakefile` to support `rake build` and `rake test`.

> We need a guide or talk in order to understand all changes and see how to pass from current stable code (in master) to this new version.

As the current version lacks of documentation and usage out of the @bq/apps environment, I think a talk would be easier to manage. For the rest, the README file has a pretty usage guide and a fine amount of tests of how to use the different parts of the architecture. I don't deny the addition of an Example Project in further PRs if needed 😄 .

> I think this version should be v1.0.0

It is, sir!

> Codecov don't register coverage for every platform anymore. The script should be executed after pass the test for every platform.

It should be doing it by now.

> pod repo update step is necessary on travis? his took a huge amount of seconds to resolve.

Removed from the `travis.yml` file!
